### PR TITLE
Add a flag to change the web API endpoint

### PR DESCRIPTION
**Pull request by cakecatz**
**What this PR does / why we need it**:

Add a flag --web-api-endpoint to server command.
This can change the API endpoint that use to call API from the browser.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/server/server.go [hunk 1]
```diff
@@ -286,12 +292,18 @@ func (s *server) run(ctx context.Context, t cli.Telemetry) error {
...
+			script = fmt.Sprintf("<script>window.API_ENDPOINT = '%s';</script>", s.webAPIEndpoint)
+		}
+
 		mux.Handle("/assets/", gziphandler.GzipHandler(assetsHandler))
 		mux.HandleFunc("/favicon.ico", func(w http.ResponseWriter, r *http.Request) {
 			http.ServeFile(w, r, filepath.Join(s.staticDir, "favicon.ico"))
 		})
 		mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
-			http.ServeFile(w, r, filepath.Join(s.staticDir, "/index.html"))
+			tmpl.Execute(w, script)
```
Comment (cakecatz) [inline cmd/server/server.go:306]: `script` is HTML code, so I use `text/template` instead of `html/template`.

File: cmd/server/server.go [hunk 2]
```diff
@@ -286,12 +292,18 @@ func (s *server) run(ctx context.Context, t cli.Telemetry) error {
 			http.StripPrefix("/assets/", fs).ServeHTTP(w, r)
 		})
 
+		tmpl := template.Must(template.ParseFiles(filepath.Join(s.staticDir, "/index.html")))
+		script := ""
+		if s.webAPIEndpoint != "" {
+			script = fmt.Sprintf("<script>window.API_ENDPOINT = '%s';</script>", s.webAPIEndpoint)
+		}
```
Comment (nghialv) [inline cmd/server/server.go:299]:
```
tmpl := template.New("index.html")
tmpl, err = tmpl.ParseFiles(filepath.Join(s.staticDir, "/index.html"))
if err != nil {
    t.Logger.Error("failed to parse index.html", zap.Error(err))
   return err
}

var data string
if s.webAPIEndpoint != "" {
 data = ...
}

var indexBuf bytes.Buffer
if err := tmpl.Execute(&indexBuf, data); err != nil {
    t.Logger.Error("failed to render index.html", zap.Error(err))
    return err
}

...

w.Write(tmpl.Bytes())
```

File: cmd/server/server.go [hunk 3]
```diff
@@ -117,6 +121,8 @@ func NewCommand() *cobra.Command {
 	cmd.Flags().StringVar(&s.configFile, "config-file", s.configFile, "The path to the configuration file.")
 	cmd.MarkFlagRequired("config-file")
 
+	cmd.Flags().StringVar(&s.webAPIEndpoint, "web-api-endpoint", s.webAPIEndpoint, "The endpoint to used to call API from web browser.")
```
Comment (nghialv) [inline cmd/server/server.go:124]: nit: "The endpoint of the web API"

File: cmd/server/server.go (hunk 2 above)
Comment (cakecatz) [reply cmd/server/server.go:312]: I changed to this.
Thank you.

**Comment by nghialv** [comment]
Thank you.
/approve

**Pull request merged**